src/test/scala/chisel3/tests/ValidQueueTest.scala

```diff
@@ -65,24 +63,49 @@ class ValidQueueTest extends FlatSpec with ChiselScalatestTester {
     }
   }
 
-  it should "wait for data until valid data presented" in {
-    test(new ValidQueueModule(UInt(8.W), delay = 3)) { c =>
-      c.in.initSource().setSourceClock(c.clock)
-      c.out.initSink().setSinkClock(c.clock)
-
-      for(_ <- 0 until 10) {
-        c.out.expectInvalid()
-        c.clock.step()
+  it should "expectInvalid can be used to wait for data" in {
+    for(delay <- 1 until 4) {
+      test(new ValidQueueModule(UInt(8.W), delay = delay)) { c =>
+        c.in.initSource().setSourceClock(c.clock)
+        c.out.initSink().setSinkClock(c.clock)
+
+        for (_ <- 0 until 10) {
+          c.out.expectInvalid()
+          c.clock.step()
+        }
+
+        for (value <- 70 to 80) {
```

Comment:
This doesn't pipeline data, is that intentional?
If you want to pipeline data, consider rewriting using threads.

Reply:
Did not want to pipleline data. just wanted to show that expectInvalid works.